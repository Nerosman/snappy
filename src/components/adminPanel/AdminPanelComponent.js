import React, {Component} from 'react';
import {Col, Row, Container} from 'reactstrap';
import {AdminPanelTableComponent} from "./AdminPanelTableComponent";
import {getOrderListRequest} from "../../communicator";

export class AdminPanelComponent extends Component {
    constructor(props) {
        super(props);
        this.state = {
            orderList: []
        }
    }

    componentDidMount() {
        getOrderListRequest().then(response => response.json())
            .then((orderList) => {
                return this.setState({
                    orderList
                })
            })
    }

    render() {
        return <Container fluid>
            <Row lg={12}>
                <Col lg={{size: 10, offset: 1}} className="well">
                    <AdminPanelTableComponent orderList={this.state.orderList}/>
                </Col>
            </Row>
        </Container>
    }
}