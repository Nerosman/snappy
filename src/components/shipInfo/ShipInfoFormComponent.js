import React from "react";
import {InputComponent} from "../reusableComponents/InputComponent";
import {Button, Form} from "reactstrap";
import {validateAddress, validateEmail, validateLengthOfValue} from "../../utils/validations";
import {toast} from "react-toastify";
import {addOrderRequest} from "../../communicator";

export class ShipInfoFormComponent extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            emailValidity: true,
            phoneValidity: true,
            specialNotesValidity: true,
            addressValidity: true,
            address1Validity: true,
            address2Validity: true,
            firstNameValidity: true,
            secondNameValidity: true,
            address1: "",
            address2: "",
            firstName: "",
            secondName: "",
            phone: "",
            specialNotes: "",
            email: ""

        }
    }

    inputValueToState(e) {
        this.setState({[e.target.name]: e.target.value})
    }

    emailParameterToState(e) {
        const email = e.target.value;
        const emailValidity = validateEmail(email);
        this.setState({
            [e.target.name]: email,
            emailValidity
        });
    }

    phoneParameterToState(e) {
        const phone = e.target.value;
        const phoneValidity = validateLengthOfValue(phone, 10);
        this.setState({
            [e.target.name]: phone,
            phoneValidity
        });
    }

    specialNotesToState(e) {
        const specialNotes = e.target.value;
        const specialNotesValidity = validateLengthOfValue(specialNotes, 500);
        this.setState({
            [e.target.name]: specialNotes,
            specialNotesValidity
        });
    }

    submitForm(e) {
        e.preventDefault();
        // validateAddress(this.state.address1, this.state.address2)
        //     .then(response => response.json())
        //     .then(jsondata => {
        //         this.setState({
        //             addressValidity: jsondata.ErrorCode === 0,
        //             addressErrorMessage: jsondata.ErrorMessage
        //         })
        //     })
        let inputsValidity = [];

        for (let key in this.state) {
            if (key.includes("Validity")) {
                inputsValidity.push(this.state[key])
            }
        }

        const isFormNonValid = inputsValidity.includes(false);

        if (isFormNonValid) {
            toast("Something went wrong. Please check your data", {
                position: toast.POSITION.BOTTOM_RIGHT,
                type: "error"
            })
        } else {
            addOrderRequest(this.state).then((res) => {
                res.status === 200 && toast("Your date has been save successfully", {
                    position: toast.POSITION.BOTTOM_RIGHT,
                    type: "success"
                })
            })
        }
    }

    render() {
        return <Form onSubmit={this.submitForm.bind(this)}>
            <InputComponent
                id="first-name"
                name="firstName"
                label="First name"
                type="text"
                onChange={this.inputValueToState.bind(this)}
                onValid={this.state.firstNameValidity}
                placeholder="John"
                errorMessage={"Input cannot be empty"}
                required={true}
            />
            <InputComponent
                id="second-name"
                name="secondName"
                label="Second name"
                type={"text"}
                onChange={this.inputValueToState.bind(this)}
                onValid={this.state.secondNameValidity}
                placeholder="Whiskey"
                errorMessage={"Input cannot be empty"}
                required={true}
            />
            <InputComponent
                id="address"
                name="address1"
                label="Address"
                type={"text"}
                onChange={this.inputValueToState.bind(this)}
                onValid={this.state.addressValidity && this.state.address1Validity}
                placeholder="1234 Main St"
                errorMessage={this.state.addressErrorMessage || "Input cannot be empty"}
                required={true}
            />
            <InputComponent
                id="cityStateZip"
                name="address2"
                label="City, State, Zip"
                type="text"
                onChange={this.inputValueToState.bind(this)}
                onValid={this.state.addressValidity && this.state.address2Validity}
                placeholder="ASBURY PARK, NJ 07712-6086 "
                errorMessage={this.state.addressErrorMessage || "Input cannot be empty"}
                required={true}
            />
            <InputComponent
                id="email"
                name="email"
                label="Email"
                type={"email"}
                onChange={this.emailParameterToState.bind(this)}
                onValid={this.state.emailValidity}
                errorMessage="Please include @ and . in the email address"
                required={true}
            />
            <InputComponent
                id="phone"
                name="phone"
                label="Phone"
                onChange={this.phoneParameterToState.bind(this)}
                onValid={this.state.phoneValidity}
                errorMessage="Length of phone number must be less then 10 charts"
                required={true}
            />
            <InputComponent
                id="special-notes"
                name="specialNotes"
                label="Special notes"
                type={"textarea"}
                onChange={this.specialNotesToState.bind(this)}
                onValid={this.state.specialNotesValidity}
                errorMessage="Length of special notes must be less then 500  charts"
            />
            <Button id="submit-ship-info-button" type="submit">Submit</Button>
        </Form>
    }
}
