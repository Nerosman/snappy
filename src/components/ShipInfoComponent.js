import React from "react";
import {Container} from "reactstrap";
import {ShipInfoFormComponent} from "./ShipInfoFormComponent";

export const ShipInfoComponent = () => {
    return <Container>
        <ShipInfoFormComponent/>
    </Container>
}