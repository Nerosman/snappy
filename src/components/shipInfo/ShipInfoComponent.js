import React from "react";
import {Col, Container, Row} from "reactstrap";
import {ShipInfoFormComponent} from "./ShipInfoFormComponent";

export const ShipInfoComponent = () => {
    return <Container fluid>
        <Row lg={12}>
            <Col lg={{ size: 6, offset: 3 }} className="well">
                <legend>
                    Fill shipping info to receive a gift
                </legend>
                <ShipInfoFormComponent/>
            </Col>
        </Row>
    </Container>
}