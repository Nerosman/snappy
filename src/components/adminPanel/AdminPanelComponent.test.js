import ReactDOM from "react-dom";
import React from "react";
import {AdminPanelComponent} from "./AdminPanelComponent";
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {ShipInfoComponent} from "../shipInfo/ShipInfoComponent";
import jest from "jest-mock";

configure({ adapter: new Adapter() });

describe("component <AdminPanelComponent />", ()=>{
    it('render div into AdminPanelComponent without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<AdminPanelComponent />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('snapshot adminPanel', () => {
        const wrapper = shallow(<AdminPanelComponent/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('check if component has one Container', () => {
        const wrapper = shallow(<AdminPanelComponent/>);
        expect(wrapper.find('Container').length).toBe(1);
    });

    it('could save orderList to state', ()=> {
        const wrapper = shallow(<AdminPanelComponent />);
        const orderList = [{firstName: "Bohdan", secondName: "Yuzyfyshyn", address1: "Kn. Olgy 12", address2: "Lviv", email: "nerosman2601@gmail.com", phone: "0937680642"}];
        const instance = wrapper.instance();
        jest.spyOn(instance, 'setOrderListToState');
        instance.setOrderListToState(orderList);
        expect(instance.setOrderListToState).toHaveBeenCalled();
        expect(wrapper.state('orderList')).toEqual(orderList);
    });
});
