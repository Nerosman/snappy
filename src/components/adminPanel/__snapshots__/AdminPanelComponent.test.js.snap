// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <AdminPanelComponent /> snapshot adminPanel 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminPanelComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Row
        lg={12}
        tag="div"
      >
        <Col
          className="well"
          lg={
            Object {
              "offset": 1,
              "size": 10,
            }
          }
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <AdminPanelTableComponent
            orderList={Array []}
          />
        </Col>
      </Row>,
      "fluid": true,
      "tag": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Col
          className="well"
          lg={
            Object {
              "offset": 1,
              "size": 10,
            }
          }
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <AdminPanelTableComponent
            orderList={Array []}
          />
        </Col>,
        "lg": 12,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <AdminPanelTableComponent
            orderList={Array []}
          />,
          "className": "well",
          "lg": Object {
            "offset": 1,
            "size": 10,
          },
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "orderList": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Row
          lg={12}
          tag="div"
        >
          <Col
            className="well"
            lg={
              Object {
                "offset": 1,
                "size": 10,
              }
            }
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <AdminPanelTableComponent
              orderList={Array []}
            />
          </Col>
        </Row>,
        "fluid": true,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Col
            className="well"
            lg={
              Object {
                "offset": 1,
                "size": 10,
              }
            }
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <AdminPanelTableComponent
              orderList={Array []}
            />
          </Col>,
          "lg": 12,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <AdminPanelTableComponent
              orderList={Array []}
            />,
            "className": "well",
            "lg": Object {
              "offset": 1,
              "size": 10,
            },
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "orderList": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
